__tests__/authutil.test.ts

```diff
@@ -1,67 +1,123 @@
-import * as io from '@actions/io';
+import os = require('os');
 import * as fs from 'fs';
 import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
 import * as auth from '../src/authutil';
 
 let rcFile: string;
 
-describe('installer tests', () => {
+describe('authutil tests', () => {
+  const _runnerDir = path.join(__dirname, 'runner');
+
+  let cnSpy: jest.SpyInstance;
+  let logSpy: jest.SpyInstance;
+  let dbgSpy: jest.SpyInstance;
+
   beforeAll(async () => {
-    const tempDir = path.join(
-      __dirname,
-      'runner',
-      path.join(
-        Math.random()
-          .toString(36)
-          .substring(7)
-      ),
-      'temp'
+    const randPath = path.join(
+      Math.random()
+        .toString(36)
+        .substring(7)
     );
+    const tempDir = path.join(_runnerDir, randPath, 'temp');
     await io.rmRF(tempDir);
     await io.mkdirP(tempDir);
     process.env['GITHUB_REPOSITORY'] = 'OwnerName/repo';
     process.env['RUNNER_TEMP'] = tempDir;
     rcFile = path.join(tempDir, '.npmrc');
   }, 100000);
 
-  beforeEach(() => {
-    if (fs.existsSync(rcFile)) {
-      fs.unlinkSync(rcFile);
-    }
+  beforeEach(async () => {
+    await io.rmRF(rcFile);
+    // if (fs.existsSync(rcFile)) {
+    //   fs.unlinkSync(rcFile);
+    // }
     process.env['INPUT_SCOPE'] = '';
-  });
+
+    // writes
+    cnSpy = jest.spyOn(process.stdout, 'write');
+    logSpy = jest.spyOn(console, 'log');
+    dbgSpy = jest.spyOn(core, 'debug');
+    cnSpy.mockImplementation(line => {
+      // uncomment to debug
+      // process.stderr.write('write:' + line + '\n');
+    });
+    logSpy.mockImplementation(line => {
+      // uncomment to debug
+      // process.stderr.write('log:' + line + '\n');
+    });
+    dbgSpy.mockImplementation(msg => {
+      // uncomment to see debug output
+      // process.stderr.write(msg + '\n');
+    });
+  }, 100000);
+
+  function dbg(message: string) {
+    process.stderr.write('dbg::' + message + '::\n');
+  }
+
+  afterAll(async () => {
+    if (_runnerDir) {
+      await io.rmRF(_runnerDir);
+    }
+  }, 100000);
+
+  function readRcFile(rcFile: string) {
```

Comment:
when I upgraded to the new typescript compiler and jest version the snapshot stuff was providing more problems than value.  I got it working by updating the baselines then after changes and updating it stopped working on CI even though I validated with debug code that it was equivalent byte for byte.  It was saying I need to update baselines on CI server even though I did.  In the end, comparing a couple keys programmatically was much simpler.

Reply:
+1 i'm honestly glad to see it deleted - one less thing to grok and i never fully understood how it worked.